src/main/java/dk/sdu/se_f22/productmodule/management/ProductManager.java

```diff
@@ -0,0 +1,356 @@
+
+        //Adds new product to the productArray.
+        //Returns whether this was possible or not.
+
+        checkForUpdates();
+        boolean success = productArray.add(p);
+        updateSource();
+
+        return success;
+    }
+
+    @Override
+    public boolean createAll(ArrayList<Product> pList){
+        checkForUpdates();
+
+        boolean success = productArray.addAll(pList);
+
+        updateSource();
+        return success;
+    }
+
+    @Override
+    public Product read(String productId) {
+
+        //This function returns a single product based on the UUID
+        //Since the read() function doesn't alter any attribute values on the product
+        //There's no reason to update the source.
+
+        checkForUpdates();
+
+        Product toReturn = null;
+
+        for(Product p : productArray){
+            if(p.get(ProductAttribute.UUID).equalsIgnoreCase(productId)){
+                toReturn = p;
+                break;
+            }
+        }
+
+        return toReturn;
+    }
+
+    @Override
+    public Product[] readAll(String[] productIds) {
+
+        //This function returns an array of products based on an array of UUID's
+        //The size of the return array should equal the size of the input ID array
+
+        checkForUpdates();
+
+        Product[] returnArray = new Product[productIds.length];
+
+        for(int i = 0; i < productIds.length; i++){
+            for(Product p : productArray){
+
+                if(p.get(ProductAttribute.ID).equalsIgnoreCase(productIds[i])){
+                    returnArray[i] = p;
+                    break;
+                }
+            }
+        }
+
+        return returnArray;
+    }
+
+    @Override
+    public boolean update(String productId, ProductAttribute a, String s) {
+
+        //This function sets the value (String) of an attribute on a product
+
+        checkForUpdates();
+
+        boolean succes = false;
+
+        for(Product pT : productArray){
+            if(Objects.equals(pT.get(ProductAttribute.UUID), productId)){
+                succes = pT.set(a,s);
+                break;
+            }
+        }
+
+        updateSource();
+
+        return succes;
+    }
+
+    @Override
+    public boolean update(String productId, Product p) {
+
+        //This function updates the attributes of a product based on its UUID
+        //By replacing it entirely with a new one. Make sure this new product has all it's attributes set correctly.
+        //Any missing attribute on this other product is also removed from the source file.
+
+        checkForUpdates();
+
+        boolean success = false;
+
+        for(Product pT : productArray){
+            if(pT.get(ProductAttribute.UUID).equalsIgnoreCase(productId)){
+                pT = p;
+                success = true;
+                break;
+            }
+        }
+
+        updateSource();
+
+        return success;
+    }
+
+    @Override
+    public boolean remove(String productId) {
```

Comment:
Use type of `UUID`

Reply:
It makes the Product more complicated if we do